job/job.go

```diff
@@ -86,3 +89,35 @@ func GetGenericJob(obj Object, config Config) (*batchv1.Job, error) {
 
 	return job, err
 }
+
+func getResourceRequirementsTemplate(r *corev1.ResourceRequirements) corev1.ResourceRequirements {
+	if r == nil {
+		return corev1.ResourceRequirements{
+			Requests: getDefaultResourceRequests(),
+			Limits:   getDefaultResourceLimits(),
+		}
+	}
+	if r.Limits == nil {
+		r.Limits = getDefaultResourceLimits()
+	}
+	if r.Requests == nil {
+		r.Requests = getDefaultResourceRequests()
+	}
+	return *r
+}
+
+func getDefaultResourceLimits() map[corev1.ResourceName]resource.Quantity {
+	li := make(map[corev1.ResourceName]resource.Quantity)
+	li[corev1.ResourceCPU] = resource.MustParse(cfg.Config.GlobalCPUResourceLimit)
+	li[corev1.ResourceMemory] = resource.MustParse(cfg.Config.GlobalMemoryResourceLimit)
```

Comment:
I see no advantage in abbreviating `limits` to `li`, it's less readable IMO (clean code).